src/core/indexedlist.h

```diff
@@ -13,30 +13,91 @@ class IndexedList
     explicit IndexedList() = default;
```

Comment:
IndexedList could be replaced by a map or a hash table, the only difference is the performance and memory usage, for our usage it easily beats the hash table or map on both points.
That said performance isn't really crucial here, so I'm not against removing it if people find it too complicated to deal with.